crates/matrix-sdk-crypto/src/store/crypto_store_wrapper.rs

```diff
@@ -109,10 +120,71 @@ impl CryptoStoreWrapper {
             // Mapping the devices and user identities from the read-only variant to one's
             // that contain side-effects requires our own identity. This is
             // guaranteed to be up-to-date since we just persisted it.
-            let own_identity =
+            let maybe_own_identity =
                 self.store.get_user_identity(&self.user_id).await?.and_then(|i| i.into_own());
 
-            let _ = self.identities_broadcaster.send((own_identity, identities, devices));
+            // If our identity was not verified before the change and is now, that means
```

Comment:
This piece of code very much feels like a hack, why aren't we doing the same thing we're doing for other user's identities?

Reply:
I am not sure to get it. For other identities if there is an update we just need to check the updated identity. But when my own identity trust state change then all the other identities signed by it would be affected. Is this what you mean? 
Also I want only to detect the precise moment the own identity goes from unverified to verified, to only do the checking of all other identities when actually needed.